server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -250,21 +214,12 @@ protected void registerAsReplacedHandler(
         RestHandler handler,
         RestRequest.Method replacedMethod,
         String replacedPath,
-        RestApiVersion replacedVersion
+        RestApiVersion replacedVersion,
+        String replacedMessage,
+        Level deprecationLevel
     ) {
-        // e.g. [POST /_optimize] is deprecated! Use [POST /_forcemerge] instead.
-        final String replacedMessage = "["
```

Comment:
this was moved in to the builder since that is the only place we can express "replaced"